app/navigation/BusinessSettlorNavigator.scala

```diff
@@ -36,46 +37,73 @@ class BusinessSettlorNavigator @Inject()() extends Navigator {
+  private def conditionalNavigation(mode: Mode, trustType: TypeOfTrust): PartialFunction[Page, UserAnswers => Call] = {
+    case NamePage => ua =>
+      navigateAwayFromNamePage(mode, ua)
     case UtrYesNoPage => ua =>
-      yesNoNav(ua, UtrYesNoPage, rts.UtrController.onPageLoad(mode), rts.AddressYesNoController.onPageLoad(mode))
+      yesNoNav(ua, UtrYesNoPage, rts.UtrController.onPageLoad(mode), navigateAwayFromUtrPages(mode, trustType, ua))
+    case UtrPage => ua =>
+      navigateAwayFromUtrPages(mode, trustType, ua)
+    case CountryOfResidenceYesNoPage => ua =>
+      yesNoNav(ua, CountryOfResidenceYesNoPage, rts.CountryOfResidenceInTheUkYesNoController.onPageLoad(mode), navigateAwayFromResidencePages(mode, trustType, ua))
+    case CountryOfResidenceInTheUkYesNoPage => ua =>
+      yesNoNav(ua, CountryOfResidenceInTheUkYesNoPage, navigateAwayFromResidencePages(mode, trustType, ua), rts.CountryOfResidenceController.onPageLoad(mode))
+    case CountryOfResidencePage => ua =>
+      navigateAwayFromResidencePages(mode, trustType, ua)
+    case AddressYesNoPage => ua =>
+      yesNoNav(ua, AddressYesNoPage, rts.LiveInTheUkYesNoController.onPageLoad(mode), navigateToEndPages(mode, trustType, ua))
     case LiveInTheUkYesNoPage => ua =>
       yesNoNav(ua, LiveInTheUkYesNoPage, rts.UkAddressController.onPageLoad(mode), rts.NonUkAddressController.onPageLoad(mode))
+    case UkAddressPage | NonUkAddressPage => ua =>
+      navigateToEndPages(mode, trustType, ua)
+    case CompanyTimePage => ua =>
+      navigateAwayFromCompanyTimePage(mode, ua)
+  }
+
+
+  private def navigateAwayFromNamePage(mode: Mode, answers: UserAnswers): Call = {
+    if (answers.is5mldEnabled && !answers.isTaxable) {
+      rts.CountryOfResidenceYesNoController.onPageLoad(mode)
+    } else {
+      rts.UtrYesNoController.onPageLoad(mode)
+    }
+  }
+
+  private def navigateAwayFromUtrPages(mode: Mode, trustType: TypeOfTrust, answers: UserAnswers): Call = {
+    (answers.is5mldEnabled, isUtrDefined(answers)) match {
+      case (true, _) => rts.CountryOfResidenceYesNoController.onPageLoad(mode)
+      case (false, true) => navigateToEndPages(mode, trustType, answers)
+      case (false, _) => rts.AddressYesNoController.onPageLoad(mode)
+    }
+  }
+
+  private def navigateAwayFromResidencePages(mode: Mode, trustType: TypeOfTrust, answers: UserAnswers): Call = {
+    val isNonTaxable5mld = answers.is5mldEnabled && !answers.isTaxable
+
+    if (isNonTaxable5mld || isUtrDefined(answers)){
+      navigateToEndPages(mode, trustType, answers)
+    } else {
+     rts.AddressYesNoController.onPageLoad(mode)
+    }
+  }
+
+  private def navigateToEndPages(mode:Mode, trustType: TypeOfTrust, ua: UserAnswers): Call = {
+    (mode, trustType) match {
+      case (_, EmployeeRelated) => rts.CompanyTypeController.onPageLoad(mode)
+      case (NormalMode, _) => rts.StartDateController.onPageLoad()
+      case (CheckMode, _) => checkDetailsRoute(ua)
```

Comment:
This could now be simplified to `if (trustType == EmployeeRelated) rts.CompanyTypeController.onPageLoad(mode) else navigateAwayFromCompanyTimePage(mode, ua)`, but I would perhaps consider renaming `navigateAwayFromCompanyTimePage` to something like `navigateToStartDateOrCheckDetails` so the name would make more sense in the context of both its uses

Reply:
Ok, makes sense. It's all getting a little bit mazey, but still probably easy enough to follow